dace/transformation/dataflow/tasklet_fusion.py

```diff
@@ -0,0 +1,226 @@
+# Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
+""" Contains classes that fuse Tasklets """
+
+from typing import Any, Dict, Set
+import ast
+import dace
+import re
+from dace import dtypes, registry
+from dace.sdfg import nodes
+from dace.sdfg import utils as sdutil
+from dace.transformation import transformation as pm
+
+
+class ConnectorRenamer(ast.NodeTransformer):
+    """ Renames connector names in Tasklet code.
+    """
+    def __init__(self, repl_dict: Dict[str, str]) -> None:
+        """ Initializes AST transformer.
+            :param repl_dict: Replacement dictionary.
+        """
+        self.repl_dict = repl_dict
+
+    def visit_Name(self, node: ast.Name) -> Any:
+        # Rename connector
+        if node.id in self.repl_dict:
+            node.id = self.repl_dict[node.id]
+        return self.generic_visit(node)
+
+
+class PythonRHSExtractor(ast.NodeVisitor):
+    """ Extracts assignments' RHS in Tasklet code.
+    """
+    def __init__(self):
+        self.assignments = set()
+
+    def visit_Assign(self, node: ast.Assign):
+        if isinstance(node.targets[0], ast.Name):
+            self.assignments.add(node.targets[0].id)
+
+    def visit_AugAssign(self, node: ast.AugAssign):
+        if isinstance(node.targets[0], ast.Name):
+            self.assignments.add(node.targets[0].id)
+
+
+@registry.autoregister_params(singlestate=True, strict=False)
+class SimpleTaskletFusion(pm.Transformation):
+    """ Fuses two connected Tasklets.
+        It is recommended that this transformation is used on Tasklets that
+        contain only simple assignments.
+    """
```

Comment:
Put an example of the before and after code (including memlets) here to help explain what happens